models/multimeter.cpp

```diff
@@ -91,27 +98,55 @@ nest::Multimeter::Parameters_::set( const DictionaryDatum& d,
   const Buffers_& b )
 {
   if ( b.has_targets_
-    && ( d->known( names::interval ) || d->known( names::record_from ) ) )
+    && ( d->known( names::interval ) || d->known( names::offset )
+         || d->known( names::record_from ) ) )
+  {
     throw BadProperty(
-      "The recording interval and the list of properties to record "
-      "cannot be changed after the multimeter has been connected to "
-      "nodes." );
+      "The recording interval, the interval offset and the list of properties "
+      "to record "
+      "cannot be changed after the multimeter has been connected to nodes." );
+  }
 
   double v;
   if ( updateValue< double >( d, names::interval, v ) )
   {
     if ( Time( Time::ms( v ) ) < Time::get_resolution() )
+    {
       throw BadProperty(
-        "The sampling interval must be at least as long "
-        "as the simulation resolution." );
+        "The sampling interval must be at least as long as the simulation "
+        "resolution." );
+    }
 
     // see if we can represent interval as multiple of step
     interval_ = Time::step( Time( Time::ms( v ) ).get_steps() );
     if ( std::abs( 1 - interval_.get_ms() / v ) > 10
         * std::numeric_limits< double >::epsilon() )
+    {
+      throw BadProperty(
+        "The sampling interval must be a multiple of the simulation "
+        "resolution" );
+    }
+  }
+
+  if ( updateValue< double >( d, names::offset, v ) )
+  {
+    // only throw error if given offset is > 0., otherwise it's not an error
+    if ( v > 0 && Time( Time::ms( v ) ) < Time::get_resolution() )
```

Comment:
`0 < v < t_resolution` is an error, because if `v` is other than the default value (0), it has to be multiple of the resolution (this means positive and at least as large as the resolution itself.)

`v > 0` was wrong, should have been `v != 0`. Not sure about introducing another variable `v_time `, or creating a new object by converting to Time(), as at that moment `v` is still double, and we're not comparing it against another object, just 0.